release/release_gold.sh

```diff
@@ -45,6 +45,25 @@ function check_usage {
 	LIFERAY_COMMON_LOG_DIR="${_PROMOTION_DIR%/*}"
 }
 
+function commit_to_branch {
```

Comment:
@kiwm, since this function is performing 2 actions, can we rename it to `commit_to_branch_and_send_pull_request`?